File: YuGiOh2/Cards/C12470447.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 邪恶的仪式
    /// </summary>
    public class C12470447
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown) ||
                player.Enemy.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown);
        }

        public static void ProcessEffect(Player player)
        {
            Change(player);
            Change(player.Enemy);
        }

        private static void Change(Player player)
        {
            foreach (var monster in player.Field.MonsterFields)
            {
                if (monster == null || monster.Status.FaceDown)
                    continue;
                monster.Status.DefensePosition = !monster.Status.DefensePosition;
            }
        }
    }
}

File: YuGiOh2/Scripts/C12470447.lua
Type = 0;
--邪恶的仪式
function checkIfAvailable(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    return false;
end

function processEffect(player)
    change(player);
    change(player.Enemy);
end

function change(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            monster.Status.DefensePosition = not monster.Status.DefensePosition;
            monster.Status.CanChangePosition = false;
        end
    end
end